### PR TITLE
Use shallow fetch and clone

### PR DESCRIPTION
**Pull request by kriskowal**
This change mitigates long running idl commands that are synchronizing all branches and tags with the remote registry/repository.

It incidentally removes test support for Node.js 0.10 since npm hard crashes with the unsupported Node.js version, and updates the repository location. Tests depend on the git remote.

### CODE REVIEW

File: test/idl.js [hunk 1]
```diff
@@ -802,6 +802,9 @@ TestCluster.test('run `idl update`', {
 
 TestCluster.test('run `idl update` with corrupt meta.json', {
 }, function t(cluster, assert) {
+    assert.end(null, 'skip')
```
Comment (Matt-Esch) [inline test/idl.js]: Was it intentional to effectively remove this test?

Comment (kriskowal) [inline test/idl.js]: Revisited. This was unrelated and failing before I made my changes so I punted. Closer investigation reveals that the error message out of `read-json` got more specific and broke this test. Fixed.

**Comment by malandrew** [comment]
Once tests pass, LGTM.

File: .travis.yml [hunk 2]
```diff
@@ -6,7 +6,6 @@ node_js:
   - "4.1"
   - "4.0"
   - "0.12"
-  - "0.10"
```
Comment (breerly) [inline .travis.yml]: Suggest moving this to separate PR
